src/cargo/ops/cargo_doc.rs

```diff
@@ -77,14 +81,38 @@ pub fn doc(ws: &Workspace<'_>, options: &DocOptions<'_>) -> CargoResult<()> {
         if path.exists() {
             let mut shell = options.compile_opts.config.shell();
             shell.status("Opening", path.display())?;
+            open_docs(&path, &mut shell)?;
+        }
+    }
+
+    Ok(())
+}
+
+fn open_docs(path: &Path, shell: &mut Shell) -> CargoResult<()> {
+    match std::env::var("BROWSER") {
```

Comment:
Note that now cargo checks `BROWSER` in all operating systems.

Reply:
I think that this will want to be `var_os` to avoid the `NotUnicode` case. Additionally could the documentation note that `$BROWSER` is respected?

Reply:
:+1: done